components/bl-draggable-list-component/src/list.js

```diff
@@ -0,0 +1,82 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+
+import { ListContent } from './list-content';
+import { AddItem } from './add-item';
+import { prepareContext, ContextTypes } from './helpers/context';
+import { enrichWithUids } from './helpers/id';
+
+export function List({ component, eventHandlers }) {
+  const { onDrag, onDelete } = eventHandlers;
+  const { itemsList, allowAdd } = component;
+  const draggedItem = useRef(null);
+  const previousList = useRef(null);
+  const [isAdded, setIsAdded] = useState(false);
+  const [data, setData] = useState([]);
+
+  useEffect(() => {
+    if (itemsList) {
+      const listWithUid = enrichWithUids(itemsList);
+      setData(listWithUid);
+    }
+  }, [itemsList]);
+
+  const dragStartHandler = useCallback((e, index) => {
+    draggedItem.current = data[index];
+    previousList.current = data;
+
+    e.dataTransfer.effectAllowed = 'move';
+    e.dataTransfer.setData('text/html', e.target.parentNode);
+    e.dataTransfer.setDragImage(e.target.parentNode, 20, 20);
+  }, [data, draggedItem]);
+
+  const dragOverHandler = useCallback((e, index) => {
+    e.preventDefault();
+    const draggedOverItem = data[index];
+
+    if (draggedItem.current.uid === draggedOverItem.uid) {
```

Comment:
where there no lot of code much better to wrap the code with the condition instead of returning void

```
if (draggedItem.current.uid !== draggedOverItem.uid) {   
    const items = data.filter(item => item.uid !== draggedItem.current.uid);
    items.splice(index, 0, draggedItem.current);

    setData(items);
}
```